ostd/src/timer/jiffies.rs

```diff
@@ -32,14 +36,36 @@ impl Jiffies {
         self.0
     }
 
+    /// Adds the given number of jiffies, saturating at [`Jiffies::MAX`] on overflow.
+    pub fn add(&mut self, jiffies: u64) {
+        self.0 = self.0.saturating_add(jiffies);
+    }
+
     /// Gets the [`Duration`] calculated from the jiffies counts.
-    pub fn as_duration(self) -> Duration {
+    fn as_duration(self) -> Duration {
         Duration::from_millis(self.0 * 1000 / TIMER_FREQ)
     }
+
+    /// Gets the jiffies counts calculated from the [`Duration`].
+    ///
+    /// Returns a `Jiffies::MAX` on overflow.
```

Comment:
It is not a good idea to return `Jiffies::MAX` on overflow. Whether this is a desirable behavior actually depends on the specific use case. This method should let the caller to decide what to do on overflow.

```rust
pub fn try_from_duration(duration: Duration) -> Option<Self> { ... }
```

But if this interface is refactored as the above form, then implementing `TryFrom<Duration>` for `Jiffies` would be more desirable.